### PR TITLE
Extend CLI commands, add new ones

### PR DESCRIPTION
**Pull request by kramimus**
New commands:
- nas grant-access
- nas revoke-access
- nas detail
- hardware status (status based on orderId rather than hardwareId)

Updated commands:
- hardware detail: parse out specific fields json format
- hardware create: create multiple servers

janky --output-json flag to dump specific parts in JSON
TODO: revert some commands to use existing --format=json flag

### CODE REVIEW

File: SoftLayer/CLI/hardware/create.py [hunk 1]
```diff
@@ -129,11 +133,15 @@ def cli(env, **args):
...
         result = mgr.place_order(**order)
 
-        table = formatting.KeyValueTable(['name', 'value'])
-        table.align['name'] = 'r'
-        table.align['value'] = 'l'
-        table.add_row(['id', result['orderId']])
-        table.add_row(['created', result['orderDate']])
-        output = table
+        if args['output_json']:
+            env.fout(json.dumps({'orderId': result['orderId']}))
```
Comment (akud) [inline SoftLayer/CLI/hardware/create.py]: It's odd that specifying json causes different content to be output, not just a different format. It looks like below you get `id: orderId, created: orderDate`, but here you get `orderId: orderId`. The key names are different and there's a missing key.

Comment (kramimus) [inline SoftLayer/CLI/hardware/create.py]: I can add created.

The id thing is kind of a headache, the SL API makes use of 2 different IDs: "orderId" and "hardwareId" in different contexts/endpoints but calls them both "id" even though the can kinda refer to the same hardware. I opted to make the orderId explicitly labeled so, so it is slightly less confusing.

File: SoftLayer/CLI/nas/detail.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import click
+import json
+
+from SoftLayer.CLI import environment
+from SoftLayer.CLI import formatting
+from SoftLayer import utils
+
+NAS_MASK = 'id,username,serviceResourceBackendIpAddress'
```
Comment (akud) [inline SoftLayer/CLI/nas/detail.py]: what does the mask do? limit the properties returned? If so I would name this `NAS_PROPERTIES`

File: SoftLayer/CLI/nas/detail.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+@click.command()
+@click.argument('nasid', type=int)
+@environment.pass_env
+def cli(env, nasid):
+    account = env.client['Account']
+    for nas in account.getNasNetworkStorage(mask=NAS_MASK):
+        if int(nas['id']) == nasid:
+            env.fout(json.dumps(nas))
```
Comment (akud) [inline SoftLayer/CLI/nas/detail.py]: `nasid` sort of implies there's only one nas with that id, so should we break here?

File: SoftLayer/managers/hardware.py [hunk 4]
```diff
@@ -395,6 +395,24 @@ def get_create_options(self):
...
+        )
+        return result
+
+    def get_hardware_id(self, order_id):
+        order = self.get_order(order_id)
+        return order\
+            ['orderTopLevelItems'][0]\
+            ['billingItem']\
+            ['provisionTransaction']\
+            ['hardwareId']
```
Comment (akud) [inline SoftLayer/managers/hardware.py]: So their whole sdk is just dicts upon dicts upon dicts? No business objects? The problem I have with this nested access is it's very easy for it to break - there's lots of keys that could potentially change if they change part of their sdk and we merge in a fix.

**Comment by akud** [comment]
looks good

**Pull request merged**